src/main/java/org/prgrms/springbootbasic/repository/FileVoucherRepository.java

```diff
@@ -0,0 +1,111 @@
+package org.prgrms.springbootbasic.repository;
+
+import java.io.EOFException;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.util.ArrayList;
+import java.util.List;
+import org.prgrms.springbootbasic.entity.FixedAmountVoucher;
+import org.prgrms.springbootbasic.entity.PercentDiscountVoucher;
+import org.prgrms.springbootbasic.entity.Voucher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Repository;
+
+@Repository
+@Profile("dev")
+public class FileVoucherRepository implements VoucherRepository {
+
+    private static final Logger logger = LoggerFactory.getLogger(FileVoucherRepository.class);
+    private static final String FIXED_VOUCHER_DB_SER = "filedb/FixedVoucherDB.ser";
+    private static final String PERCENT_VOUCHER_DB_SER = "filedb/PercentVoucherDB.ser";
+    private final File fixedVoucherStorage = new File(FIXED_VOUCHER_DB_SER);
+    private final File percentVoucherStorage = new File(PERCENT_VOUCHER_DB_SER);
+
+    @Override
+    public void save(Voucher voucher) {
+        logger.info("FileVoucherRepository.save() called");
+
+        if (voucher instanceof FixedAmountVoucher) {
+            saveVoucherToFile(fixedVoucherStorage, voucher);
+        }
+
+        if (voucher instanceof PercentDiscountVoucher) {
+            saveVoucherToFile(percentVoucherStorage, voucher);
+        }
+    }
+
+    private void saveVoucherToFile(File file, Voucher voucher) {
+        try (ObjectOutputStream stream = new ObjectOutputStream(
+            new FileOutputStream(file, true))) {
+            stream.writeObject(voucher);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public List<Voucher> findAll() {
+        logger.info("FileVoucherRepository.findAll() called");
+
+        List<Voucher> vouchers = new ArrayList<>();
+
+        try (FileInputStream fixedVoucherFileStream = new FileInputStream(
+            fixedVoucherStorage)) {
+            readFixedAmountVoucher(vouchers, fixedVoucherFileStream);
+        } catch (EOFException of) {
+        } catch (IOException | ClassNotFoundException ex) {
+            ex.printStackTrace();
+        }
```

Comment:
```suggestion
        }  catch (EOFException | IOException | ClassNotFoundException ex) {
            ex.printStackTrace();
        }
```

이렇게 사용이 블가할까요?
아래도 동일합니다.

Reply:
EOFException과 나머지 Exception을 다르게 처리하고 싶어서 분리했습니다. 
EOFException은 파일을 끝까지 읽어서 나오는 예외이기 때문에 이 경우 정상적으로 동작하게하고, 나머지 예외는 문제가 있는 경우이기 때문에 예외를 출력하게 구현했습니다.

어떻게 보면 제가 EOFException에 로깅을 남겨 두면 다른 분이 코드를 볼때 오해하지 않았을 것 같네요. 이 부분에 로깅을 남겨 제 의도를 명확하게 하겠습니다!